### PR TITLE
[Bug] Anon players should not show in /who all

### PR DESCRIPTION
**Pull request by fryguy503**
# Description

Updated to properly filter anon players from /who all.

The code formatting was very inconsistant and needed cleanup.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# Testing

The /anon player is in Gunthak for this example, other player is in Dulak

Myself in anon mode doing `/who` (Not /who all) - Same zone

image

Myself in anon mode doing `/who all gunthak`

image

Player in another zone doing `/who all gunthak`

image

Player in another zone doing `/who all strung` (Strung is the anon players name)

image

# Checklist

- [X] I have tested my changes
- [X] I have performed a self-review of my code. Ensuring variables, functions and methods are named in a human-readable way, comments are added only where naming of variables, functions and methods can't give enough context.
- [X] I own the changes of my code and take responsibility for the potential issues that occur

### CODE REVIEW

**Comment by Akkadius** [comment]
It's really hard to tell what actually changed in this PR. Formatting that is this sweeping should generally be separate from logic changes.

**Comment by fryguy503** [comment]
tl;dr it is roughly this for anon/rp coverage

```cpp
				} else if (((countcle->Anon() == 1 && admin <= countcle->Admin()) && whomlen != 0 &&
							strncasecmp(countcle->name(), whom->whom, whomlen) == 0)) {
					totalusers++;
					if (totalusers <= 20 || admin >= AccountStatus::GMAdmin) {
						totallength = totallength + strlen(countcle->name()) + strlen(countcle->AccountName()) +
									strlen(guild_mgr.GetGuildName(countcle->GuildID())) + 5;
					}
				} else if (((countcle->Anon() == 2 && admin <= countcle->Admin()) && whomlen != 0 &&
							(strncasecmp(countcle->name(), whom->whom, whomlen) == 0 ||
							strncasecmp(guild_mgr.GetGuildName(countcle->GuildID()), whom->whom, whomlen) == 0))) {
					totalusers++;
					if (totalusers <= 20 || admin >= AccountStatus::GMAdmin) {
						totallength = totallength + strlen(countcle->name()) + strlen(countcle->AccountName()) +
									strlen(guild_mgr.GetGuildName(countcle->GuildID())) + 5;
					}
				}

**Comment by Akkadius** [comment]
That helps, even with the formatting this function needs a tremendous amount of help

**Review by Akkadius — APPROVED** [review]
